src/app/api/advocates/route.ts

```
@@ -1,8 +1,38 @@
 import db from "../../../db";
 import { advocates } from "../../../db/schema";
+import { NextRequest } from "next/server";
+import { count } from "drizzle-orm";
 
-export async function GET() {
-  const data = await db.select().from(advocates);
+export async function GET(request: NextRequest) {
+  // Get URL parameters for pagination
+  const searchParams = request.nextUrl.searchParams;
+  const page = parseInt(searchParams.get("page") || "1");
+  const limit = parseInt(searchParams.get("limit") || "10");
+  
+  // Validate input parameters
+  const validatedPage = page > 0 ? page : 1;
+  const validatedLimit = limit > 0 && limit <= 100 ? limit : 10;
```

Defaulting like this is probably fine, but maybe we'd want to throw an error or at least log this to make sure no one's trying to do anything malicious.